atmo/appsource/appsource.go

```diff
@@ -13,37 +13,48 @@ var (
 	ErrAuthenticationFailed = errors.New("failed to authenticate")
 )
 
-// Meta describes the metadata for an App
+// Meta describes the metadata for an App.
 type Meta struct {
 	Identifier string `json:"identifier"`
 	AppVersion string `json:"appVersion"`
+	Domain     string `json:"domain"`
 }
 
 type AppSource interface {
-	// Start indicates to the AppSource that it should prepare for app startup
-	Start(options.Options) error
-	// Runnables returns all of the available Runnables
-	Runnables() []directive.Runnable
+	// Start indicates to the AppSource that it should prepare for app startup.
+	Start(opts options.Options) error
+
+	// Runnables returns all of the available Runnables.
+	Runnables(ident, version string) []directive.Runnable
+
 	// FindRunnable directs the AppSource to attempt to find
 	// a particular Runnable and make it available at next
 	// AppSource state sync via a call to Runnables().
 	// ErrRunnableNotFound should be returned if the Runnable cannot be found.
 	FindRunnable(fqfn, authHeader string) (*directive.Runnable, error)
```

Comment:
Hmm I wonder if we should split out the components of FQFN here to be consistent with the others.

Reply:
Should this be in the scope of this PR though? That feels like it should have its own issue.

Reply:
(In another PR) would it make sense to just make FQFN a first class type?

Reply:
> (In another PR) would it make sense to just make FQFN a first class type?

We already have the FQFN as a separate struct here: https://github.com/suborbital/atmo/blob/main/fqfn/fqfn.go#L23

I'm not sure whether it's being used anywhere, the fqfn.FromURL returns a string, error

Reply:
Nice, I'm probably missing some context but part of me would want all public APIs to accept the FQFN type as opposed to a raw String. Anyway just some food for thought.

Reply:
I am all for leveraging the use of static types with rigid formats!